src/test/java/uk/gov/hmcts/reform/civil/handler/tasks/TakeCaseOfflineHandlerTest.java

```diff
@@ -112,4 +114,33 @@ void shouldNotCallHandleFailureMethod_whenExceptionOnCompleteCall() {
             anyLong()
         );
     }
+
+    @Test
+    void shouldHandleExceptionAndContinue_whenOneCaseErrors() {
+        long caseId = 1L;
+        long otherId = 2L;
+        Map<String, Object> data = Map.of("data", "some data");
+        List<CaseDetails> caseDetails = List.of(
+            CaseDetails.builder().id(caseId).data(data).build(),
+            CaseDetails.builder().id(otherId).data(data).build());
+
+        when(searchService.getCases()).thenReturn(caseDetails);
+
+        String errorMessage = "there was an error";
+
+        doThrow(new NullPointerException(errorMessage))
+            .when(applicationEventPublisher).publishEvent(eq(new TakeCaseOfflineEvent(caseId)));
+
+        handler.execute(mockTask, externalTaskService);
+
+        verify(externalTaskService, never()).handleFailure(
```

Comment:
should we verify that it was called with both `caseId` and `otherId`?

Reply:
yep we can do that, suppose it's better to be verbose with unit tests!